BlazorApp1/wwwroot/blazor-fusionCharts.js

```diff
@@ -1,33 +1,113 @@
-            result = result.toString();
-        }
-        else if(typeofresult === "object"){
-            result = JSON.stringify(result);
-        }
-        else if(typeofresult === "XML"){
-            result = XML.stringify(result);
-        }
-        if(typeofresult === "string"){
-            return result;
-        }
-        return result.toString() || "";
-    };
\ No newline at end of file
+}
+
+function extractFunctionBody(functionString) {
+    //regex to extract the function body
+    const regex = /function[^{]*{([\s\S]*)}$/;
+    const match = regex.exec(functionString);
+
+    if (match && match.length >= 2) {
+        const functionBody = match[1].trim();
+        return functionBody;
+    } else {
+        return null; // Function body not found
+    }
+}
+
+// reviver function, key is passed by the JSON.parse
+function parseFunction(key, value) {
+    if (typeof value === "string" && value.match(/function\s*\(([^)]*)\)\s*\{/)) {
+        let fnArguments = extractArguments(value);
+        let fnBody = extractFunctionBody(value);
+        let parsedFn = fnArguments.length > 0 ? new Function(fnArguments, fnBody) : new Function(fnBody);
+        return parsedFn;
+    }
+    return value;
+}
+
+// Responsible for rendering any chart//
+window.FusionCharts.renderChart = (chartConfiguration) => {
+  const configAsJSObject = JSON.parse(chartConfiguration, this.parseFunction);
+  const chart = new FusionCharts(configAsJSObject);
+  chart.render();
+};
+
+// Method written for charts like time-series chart
+window.FusionCharts.setDataStore = (id, args) =>  {
+
+    let currentChart = FusionCharts(id);
+    let fusionDataStore = new FusionCharts.DataStore();
+    
+    let data = JSON.parse(args[0]);
+    let schema = JSON.parse(args[1]);
+
+    let fusionTable = fusionDataStore.createDataTable(data, schema);
+
+    currentChart.setChartData({data: fusionTable});
+};
+
+// resizeTo will return circular json object which cannot be stringified
+// and that's why it cannot be called using generic method
+window.FusionCharts.resizeTo = (id, args) => {
+    let currentChart = FusionCharts(id);
+    currentChart.resizeTo(args[0], args[1]);
+}
+
+//To add Annotation items and groups
+window.FusionCharts.addAnnotations = (functionName, id, args) => {
+  let annotations = FusionCharts(id).annotations;
+  annotations[functionName].apply(annotations, args);
+}
+
+//Generic Method to call any fusion chart method exluding except above methods//
+window.FusionCharts.invokeChartFunction = (functionName, chartID, ...args) => {
+    // data is an array
+    let currentChart = FusionCharts(chartID), result;
+  
+    if (args[0].type === CALLBACK) {
+
+      let { event, fn } = args[0];
+      let callbackFn = parseFunction(null, fn);
+      result = currentChart[functionName].call(currentChart, event, callbackFn);
+
+      return String(result);
+
+    } else if (args[0].returnType === "NULL") {
```

Comment:
we do not need this block, we can simply put the JSON.stringify in a try catch block

Reply:
done. Added console.log(error) in catch block.